core/src/kmx/kmx_plus.cpp

```diff
@@ -806,81 +1012,97 @@ COMP_KMXPLUS_LAYR_Helper::getRow(KMX_DWORD row) const {
 
 const COMP_KMXPLUS_LAYR_KEY *
 COMP_KMXPLUS_LAYR_Helper::getKey(KMX_DWORD key) const {
-  if (!valid() || key >= layr->keyCount) {
+  if (!valid() || key >= data()->keyCount) {
     assert(false);
     return nullptr;
   }
   return keys + key;
 }
 
 bool
-COMP_KMXPLUS_KEYS::valid(KMX_DWORD _kmn_unused(length)) const {
-  if (header.size < sizeof(*this)
+COMP_KMXPLUS_KEYS::valid(COMP_KMXPLUS_HEADER const &header, KMX_DWORD _kmn_unused(length)) const {
+  if(!is_block_valid(header, header.headerSize(),
+    sizeof(*this)
       + (keyCount    * sizeof(COMP_KMXPLUS_KEYS_KEY))
       + (flicksCount * sizeof(COMP_KMXPLUS_KEYS_FLICK_LIST))
       + (flickCount  * sizeof(COMP_KMXPLUS_KEYS_FLICK_ELEMENT))
-      + (kmapCount   * sizeof(COMP_KMXPLUS_KEYS_KMAP))) {
-    DebugLog("header.size < expected size");
-    assert(false);
+      + (kmapCount   * sizeof(COMP_KMXPLUS_KEYS_KMAP)))) {
     return false;
   }
   // further validation in the COMP_KMXPLUS_KEYS_Helper helper obj
   return true;
 }
 
 
-COMP_KMXPLUS_KEYS_Helper::COMP_KMXPLUS_KEYS_Helper() : key2(nullptr), is_valid(false) {
+COMP_KMXPLUS_KEYS_Helper::COMP_KMXPLUS_KEYS_Helper() : is_valid(false) {
 }
 
 bool
-COMP_KMXPLUS_KEYS_Helper::setKeys(const COMP_KMXPLUS_KEYS *newKeys) {
+COMP_KMXPLUS_KEYS_Helper::set(const COMP_KMXPLUS_KEYS *newKeys) {
+  COMP_KMXPLUS_Section_Helper<COMP_KMXPLUS_KEYS>::set(newKeys);
```

Comment:
if this returns false, we should abort and return false also. Need to check here and other helper::set overrides